packages/blocks/src/lib/maplibre/hooks.ts

```diff
@@ -0,0 +1,49 @@
+import { getContext, onDestroy } from 'svelte';
+import type {
+  LngLat,
+  Map,
+  MapLayerEventType,
+  MapLayerMouseEvent,
+  MapLayerTouchEvent,
+} from 'maplibre-gl';
+import { type Writable, get } from 'svelte/store';
+
+/**
+ * Provides context for a <MapLibre> instance. Must be called within a child of this component.
+ */
+export const useMapLibre = () => {
+  const mapContext = getContext<Writable<Map> | undefined>('map');
```

Comment:
We've been using [`Symbol`s](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol) with descriptive names for context keys to avoid potential collisions. e.g.

```ts
const MapContextKey = Symbol('prime-blocks-map')

// ...

const mapContext = getContext<....>(MapContextKey)
```

Reply:
Ah great suggestion, will do